### PR TITLE
tests: Collect all failures in a test derivation

### PR DESCRIPTION
**Pull request by traxys**
This runs all the test cases for a test derivation, and prints out all failures at the end, instead of stopping at the first failure

### CODE REVIEW

File: tests/test-derivation.nix [hunk 1]
```diff
@@ -38,24 +38,39 @@ let
...
-              if [[ -n $output ]]; then
-                echo "ERROR: $output"
-                exit 1
-              fi
-            ''
-          ) testList
+                output=$(HOME=$(realpath .) ${lib.getExe derivation} -mn --headless "+q" 2>&1 >/dev/null)
+                if [[ -n $output ]]; then
+                  echo "  test failure"
+                  errors[${name}]=$output
```
Comment (MattSturgeon) [inline tests/test-derivation.nix:58]: Would it be simpler to print `$output` here (and set `$status`) instead of collecting + printing later?

**Review by MattSturgeon — COMMENTED** [review]
Awesome! Seems to work, I just wonder if we can do this in a single loop?

File: tests/test-derivation.nix (hunk 1 above)
Comment (traxys) [reply tests/test-derivation.nix:58]: I'd like to avoid that, as I find it easier to look at then end of the build!

Comment (MattSturgeon) [reply tests/test-derivation.nix:58]: Ah, wasn't sure if the build was full of other output or fairly empty.

Would it make sense to print twice then? Once in context and again at the end?

Comment (MattSturgeon) [reply tests/test-derivation.nix:58]: Not pushing for this, just trying to understand the best options

Comment (traxys) [reply tests/test-derivation.nix:58]: I think you are right, in that there would be little output. I'll change that

Comment (MattSturgeon) [reply tests/test-derivation.nix:58]: I'm happy either way, assuming the output is readable. Whatever you prefer :grin:

**Review by MattSturgeon — APPROVED** [review]
Not sure if you plan on changing the implementation, but as I said I'm happy either way, here's an approving review.

**Comment by MattSturgeon** [comment]
Can we revisit this now the CI is working again?

**Comment by MattSturgeon** [comment]
I think this is obsolete now, due to other refactoring to the test suite.

**Pull request closed without merging**